src/lib/statetable.rs

```diff
@@ -246,7 +246,7 @@ impl<'a> Iterator for StateActionsIterator<'a> {
         let mut i = self.i;
         while i < self.end {
             if self.stable.used_actions.get(i).unwrap() {
-                self.i = i + 1;
+                self.i += 1;
```

Comment:
I think my initial comment was nonsense, and this line is now wrong. It was: `self.i = i + 1;` and I think `i` and `self.i` are not always the same -- right?

Reply:
Oops, you're right -- I didn't check!

Reply:
Interesting that that change still passed the unit tests though. Sorry for misleading you!